woocommerce/woocommerce
# Localize payment extension suggestions when a new country is selected

### Changes proposed in this Pull Request:

- Refresh the local redux store with new providers from the backend when a new country is selected.
- Save user choices in the backend so it can persist after refresh.





This depends on #53075
Closes #51454



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Pre-requisites for local testing:**

1. Have woocommerce installed locally and synced within WooPayments. See Pe4Hk0-3f-p2 for details.
3. Enable the feature flag to see reactified settings page. pe4Hk0-1ly-p2.
4. Check out this branch.
5. Run pnpm i && pnpm run build

**Testing default country:**

1. Make sure to delete the ` woocommerce_payments_nox_profile ` user meta
2. Change the store country to a country X
3. Go to `WooCommerce > Settings > Payments`
4. Make sure the default selected country is X (defaulting to Woo country)

**Testing changing the country:**

1. Navigate to `WooCommerce > Settings > Payments`
2. Click the Country selector at the top right, and select a new country. Click on `Apply`.
3. The menu should close and the suggestions should be refreshed with new choices.
4. Select a few non-supported countries (Morocco, Ukraine...) and make sure WooPayments isn't showing and only supported methods are displayed.
5. Refresh the page, make sure your selected country is the one selected (instead of the default WooCommerce country)



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Display payment suggestions for each selected country in WooCommerce > Settings > Payments

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
